blogs/2025/05/12/agent-mode-meets-mcp.md

```diff
@@ -0,0 +1,62 @@
+Date: 2025-05-12
+Author: Harald Kirschner
+---
+
+# Agent Mode Meets MCP: Bring Your Own Tools
+
+May 12th, 2025 by [Harald Kirschner](https://github.com/digitarald)
+
+Since MCP (Model Context Protocol) support landed last month in VS Code, it’s time for a deep dive into what it does and how it works. VS Code’s agent mode is deeply integrated to access everything inside your workspace; update files, check for errors, run build tasks, check the tests, and more. But if you ask the agent to write database queries, you might need your database schema is in your codebase or paste it in for accurate results. Once you install an MCP database server, like [postgres](https://github.com/modelcontextprotocol/servers/tree/main/src/postgres) or [sqlite](https://github.com/modelcontextprotocol/servers/tree/main/src/sqlite), agent mode can connect check your database for current tables and fields to write real, up-to-date info for queries.
+
+<!-- [ANIMATION: MCP agent workflow in VS Code] -->
+
+Think of MCP as the “API for your AI.” Instead of being stuck with whatever the model was trained on, you can wire up custom endpoints, scripts, or services, and VS Code’s agent mode can query them on the fly.  MCP is fast-growing ecosystem of interoperable building blocks to extend AI agents, so you will find servers for any use case—some examples:
+
+- [Playwright MCP Server](https://github.com/microsoft/playwright-mcp): Agent opens the web preview after making frontend changes and does a visual verification.
+
+- [GitHub MCP Server](https://github.com/github/github-mcp-server/): Agent searches for the most up-voted bug, diagnose and fix it and create a draft PR.
+
+- [Context7](https://github.com/upstash/context7/): Agent reads the developer documentation to pick the right APIs and follow framework best practices.
+
+The [official server repository](https://github.com/modelcontextprotocol/servers) is a good place to find more MCP servers.
+
+## Easy and Secure to Manage
+
+Most MCP setup instructions will ask to copy JSON objects into a mcp.json file, like .vscode/mcp.json, but we wanted to make it even easier to get started. VS Code’s **MCP: Add Server** command takes a NPM, PyPi or Docker package and automatically discover the necessary commands to set up the MCP server. Websites can show an _Install in VS Code_ button ([learn how](https://code.visualstudio.com/docs/copilot/chat/mcp-servers#_url-handler)) that opens VS Code to confirm and finish the setup. VS Code also auto-discovers existing MCP configurations from Claude Desktop or other clients, eliminating redundant setup.
+
+<!-- [MCP server quick setup in VS Code] -->
+
+Another pain point of configuring MCP servers is how to manage secrets in the configuration. We added some convenience features to keep your secrets out of source control:
+
+- **Input variables**: Inputs are prompted once when the MCP server starts the first time, then encrypted at rest when marked as _password._
```

Comment:
Not sure if how far I need to explain here or just show a screenshot; as this is also in the docs.